odoo/tools/image.py

```diff
@@ -5,8 +5,8 @@
 import io
 
 from PIL import Image, ImageOps
-# We can preload Ico too because it is considered safe
-from PIL import IcoImagePlugin
+# We can preload Ico and WebP too because they are considered safe
+from PIL import IcoImagePlugin, WebPImagePlugin
```

Comment:
Could you elaborate as to what makes you say it is safe?

Its safety would seem to rely on the safety of libwebp, which has had quite a lot of critical CVEs in the last few months.

Reply:
Odoo currently uses Pillow 8.1.2.
[Pillow 8.1.2 uses libwebp 1.2.0.](https://github.com/python-pillow/Pillow/blob/8.2.x/depends/install_webp.sh)
[Libwebp is currently at version 1.2.2.](https://chromium.googlesource.com/webm/libwebp/+/refs/tags/v1.2.2/NEWS)
As far as I understand, [CVEs on libwebp are about versions before 1.0.1.](https://www.opencve.io/cve?vendor=webmproject&product=libwebp)

Note that Pillow 9.0.x uses libwebp 1.2.1. It contains the additional hardening:
- [497](https://bugs.chromium.org/p/webp/issues/detail?id=497&q=497&can=2): happens when using the lib to convert from a PNG - which we do not do
- [508](https://bugs.chromium.org/p/webp/issues/detail?id=508&q=508&can=2): compilation warning - but was concluded as being harmless
- [518](https://bugs.chromium.org/p/webp/issues/detail?id=518&q=518&can=2): happens when using the lib to convert from an animated GIF - which we do not do

Reply:
> As far as I understand, [CVEs on libwebp are about versions before 1.0.1.](https://www.opencve.io/cve?vendor=webmproject&product=libwebp)

Yes, and that's still quite concerning. If Pillow depends on a binary lib and that lib has exploits, by exposing it we're making Odoo deployments vulnerable to attacks the next time such an exploit comes out. And it's trivial to reach it if attackers only need to upload an image to trigger it (there are many ways to do that, even without a login).

Even though libwebp is not new, the fact that fresh exploits come up shows that it's not really mature/battle-tested, and more exploits are likely to come out in the future. We have a similar problem with ghostscript and other third-party document processing libs that we can't trust.

In contrast, the `IcoImagePlugin` seems to only depend on pure-Python constructs (+ some low-level ctypes), including `BmpImagePlugin` and `PngImagePlugin` - which would make it truly "safe" against buffer overflows, etc.

Due to the above, I don't think it's fair to say that WEBP is as safe as ICO, and I would recommend that we do not enable it by default. We could have a system parameter to enable it, possibly with an explicit config setting.

Reply:
Removed the dependency on the Pillow WebP plugin.
It seems only the quality adjustment feature is lost.